drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -101,28 +102,39 @@ void ng_at86rf2xx_reset(ng_at86rf2xx_t *dev)
     /* set short and long address */
 #if CPUID_ID_LEN
     cpuid_get(cpuid);
-#if CPUID < 8
+#if CPUID_ID_LEN < 8
     /* in case CPUID_ID_LEN < 8, fill missing bytes with zeros */
     for (int i = CPUID_ID_LEN; i < 8; i++) {
         cpuid[i] = 0;
     }
+    /* copy and set long address */
+    memcpy(&addr_long, cpuid, 8);
```

Comment:
This + https://github.com/RIOT-OS/RIOT/pull/2963/files#diff-1042c565e8e083247b2102811d2f02ecR137 will result in the short address always being 0 ;-)